src/amo/components/Search/index.js

```diff
@@ -40,12 +40,29 @@ export class SearchBase extends React.Component {
     count: 0,
     enableSearchSort: true,
     filters: {},
+    paginationQueryParams: null,
     pathname: '/search/',
     results: [],
   }
 
   componentWillMount() {
-    const { dispatch, filters } = this.props;
+    this.dispatchSearch({ filters: this.props.filters });
+  }
+
+  componentWillReceiveProps({ filters: newFilters }) {
+    const { filters: oldFilters } = this.props;
+
+    if (!deepEqual(oldFilters, newFilters)) {
+      this.dispatchSearch({ filters: newFilters });
+    }
+  }
+
+  dispatchSearch({ filters = {} } = {}) {
+    const { dispatch } = this.props;
+
+    if (hasSearchFilters(filters)) {
```

Comment:
The intent was not to dispatch the search if there aren't enough filters for a (meaningful) search. I'll see about a test and also moving the logic here, maybe there's a move obvious place for it.